src/language-js/embed.js

```diff
@@ -614,8 +617,39 @@ function printHtmlTemplateLiteral(
     }
   );
 
+  const leadingWhitespace = /^\s/.test(node.quasis[0].value.raw) ? " " : "";
+  const trailingWhitespace = /\s$/.test(
+    node.quasis[node.quasis.length - 1].value.raw
+  )
+    ? " "
+    : "";
```

Comment:
```suggestion
  const [leadingWhitespace, trailingWhitespace] = [
    node.quasis[0].value.raw.slice(0, 1),
    node.quasis[node.quasis.length - 1].value.raw.slice(-1)
  ].map(char => /\s/.test(char) ? " " : "")
```

Reply:
Seems already refactor

Reply:
Not refactor, a fix for [this case](https://deploy-preview-7208--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuc0DOMAEU4bgEwDEBXKTAXkwHoAqTACQBUBZAGUxqswAMASYCMXgAnAKIAPAA7CAFAEYAlAF8AOmUwAeNGGEBLSTAB8azKcyQoGTDEkAbCpgAWMALa2VK7h49QN+XQBuxir8ulA4YlKyikqY-IIiEtIyAEzKGlT+Qd5qHtwA3CaaVNp6BoYFIAA0IBAGuujIoACGwsIQAO4ACq0IaMggzbYdzQCe-TUARsLNYADWcDAAypKzYQDmyDDCxHA1zm4A6o668GirYHBLfaeBp6MDYGgTIGFocMIwXTPrLs3IADMhu8agArNDiABCM3miyWzRccFYYTggOBexA4PESw2tjgAEViBB4GjbCCQKthO9hANJs1JnBbNUKXpYIddPgYI5kAAOAAMNWkEHehxmkgG0lwHwCqJqAEcifBvnV+ihmmgALQ4AgEZnCOAK3T677NX7-JBAskY94uXRbHbW3EExWoi3omowenszncpApD0zXS2DYAYQgLj+A1wAFZmcR3ox6arLeSArsAJJQfAIZY6fQwACCWaWMFGeNJ7yUSiAA).